grpc/resolver.go

```diff
@@ -0,0 +1,106 @@
+package grpc
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"strings"
+
+	"google.golang.org/grpc/resolver"
+)
+
+// init registers the `staticBuilder` with the gRPC resolver registry.
+func init() {
+	resolver.Register(newStaticBuilder())
+}
+
+// newStaticBuilder creates a `staticBuilder` used to construct static DNS
+// resolvers.
+func newStaticBuilder() resolver.Builder {
+	return &staticBuilder{}
+}
+
+// staticBuilder implements the `resolver.Builder` interface.
+type staticBuilder struct{}
+
+// Build implements the `resolver.Builder` interface and is usually called by
+// the gRPC dialer. It takes a target containing a comma separated list of
+// IPv4/6 addresses and a `resolver.ClientConn` and returns a `staticResolver`
+// which implements the `resolver.Resolver` interface.
+func (sb *staticBuilder) Build(target resolver.Target, cc resolver.ClientConn, _ resolver.BuildOptions) (resolver.Resolver, error) {
+	var resolverAddrs []resolver.Address
+	for _, address := range strings.Split(target.Endpoint, ",") {
+		parsedAddress, err := parseResolverIPAddress(address)
+		if err != nil {
+			return nil, err
+		}
+		resolverAddrs = append(resolverAddrs, *parsedAddress)
+	}
+	return newStaticResolver(cc, resolverAddrs), nil
+}
+
+// Scheme returns the scheme that `staticBuilder` will be registered for
+// example: `static:///`.
```

Comment:
Something about this sentence doesn't parse quite right; did you mean
```
// Scheme returns the scheme that `staticBuilder` will be registered for,
// for example: `static:///`.
```
?

Reply:
Ah, yeah, good catch.